c++/src/kj/common.h

```diff
@@ -1396,6 +1396,12 @@ class ArrayPtr: public DisallowConstCopyIfNotConst<T> {
   inline constexpr ArrayPtr(decltype(nullptr)): ptr(nullptr), size_(0) {}
   inline constexpr ArrayPtr(T* ptr, size_t size): ptr(ptr), size_(size) {}
   inline constexpr ArrayPtr(T* begin, T* end): ptr(begin), size_(end - begin) {}
+  ArrayPtr<T>& operator=(Array<T>&&) = delete;
+  ArrayPtr<T>& operator=(nullptr_t) {
```

Comment:
`nullptr_t` is a member of the `std` namespace and normally not defined if you haven't included the right header. You will need to write `decltype(nullptr)` instead (which is what `nullptr_t` is typedef'd as in the std headers).

I'm a little surprised this compiles on GCC and MSVC...